src/test/ui/rust-2018/uniform-paths/block-scoped-shadow.stderr

```diff
@@ -1,45 +1,57 @@
-error: `Foo` import is ambiguous
-  --> $DIR/block-scoped-shadow.rs:21:9
-   |
-LL | enum Foo { A, B }
-   | ----------------- can refer to `self::Foo`
-...
-LL |     enum Foo {}
-   |     ----------- shadowed by block-scoped `Foo`
+error[E0659]: `Foo` is ambiguous (name vs any other name during import resolution)
+  --> $DIR/block-scoped-shadow.rs:19:9
+   |
 LL |     use Foo::*;
-   |         ^^^
+   |         ^^^ ambiguous name
+   |
+note: `Foo` could refer to the enum defined here
+  --> $DIR/block-scoped-shadow.rs:18:5
    |
-   = help: write `self::Foo` explicitly instead
-   = note: relative `use` paths enabled by `#![feature(uniform_paths)]`
+LL |     enum Foo { A, B }
+   |     ^^^^^^^^^^^^^^^^^
+note: `Foo` could also refer to the enum defined here
+  --> $DIR/block-scoped-shadow.rs:13:1
+   |
+LL | enum Foo {}
+   | ^^^^^^^^^^^
+   = help: use `self::Foo` to refer to this enum unambiguously
 
-error: `std` import is ambiguous
-  --> $DIR/block-scoped-shadow.rs:28:9
+error[E0659]: `std` is ambiguous (name vs any other name during import resolution)
+  --> $DIR/block-scoped-shadow.rs:26:9
    |
-LL | struct std;
-   | ----------- can refer to `self::std`
-...
-LL |     enum std {}
-   |     ----------- shadowed by block-scoped `std`
 LL |     use std as foo;
-   |         ^^^ can refer to external crate `::std`
+   |         ^^^ ambiguous name
    |
-   = help: write `::std` or `self::std` explicitly instead
-   = note: relative `use` paths enabled by `#![feature(uniform_paths)]`
-
-error: `std` import is ambiguous
-  --> $DIR/block-scoped-shadow.rs:28:9
+note: `std` could refer to the enum defined here
+  --> $DIR/block-scoped-shadow.rs:25:5
    |
-LL | struct std;
-   | ----------- can refer to `self::std`
-...
-LL |     fn std() {}
-   |     ----------- shadowed by block-scoped `std`
 LL |     enum std {}
+   |     ^^^^^^^^^^^
+note: `std` could also refer to the struct defined here
+  --> $DIR/block-scoped-shadow.rs:15:1
+   |
+LL | struct std;
+   | ^^^^^^^^^^^
+   = help: use `self::std` to refer to this struct unambiguously
```

Comment:
Am I correct in understanding that `self::std` unambiguously refers to the "innermost containing module"? In other words, there is no "canonical way" to name a fn-local item, other than not being shadowed, right? And similarly, if we stabilize `use self::foo` in this way, we could never make it consider fn-local items (not saying we would want to, but it's worth noting).

Reply:
>Am I correct in understanding that self::std unambiguously refers to the "innermost containing module"?

Yes, `mod` items only.
That goes pretty deeply into the system, e.g. privacy works with `mod` granularity as well and that's why private is a sugar for `pub(self)`.

>In other words, there is no "canonical way" to name a fn-local item, other than not being shadowed, right?

RIght.

>And similarly, if we stabilize use self::foo in this way, we could never make it consider fn-local items (not saying we would want to, but it's worth noting).

Do you mean desugaring `use foo` -> `use self::foo`? (Because `use self::foo` is already stable.)
Then yes, `foo` aka `self::foo` won't be able to refer to fn-local items.